packages/space-opera/src/components/mobile_view/open_mobile_view.ts

```diff
@@ -155,87 +131,127 @@ export class OpenMobileView extends ConnectedLitElement {
+  // If a session didn't received content last time (isStale), we'll force
+  // everything to update.
+  getStaleContent(): EditorUpdates {
+    return {
+      gltfChanged: true, stateChanged: true,
+          envChanged: this.urls.env !== undefined, envIsHdr: this.envIsHdr(),
+          gltfId: getRandomInt(1e+20), usdzId: getRandomInt(1e+20),
+          iosChanged: this.urls.usdz !== undefined
     }
-    return await response.blob();
   }
 
-  async postContent(content: string|Blob, url: string) {
-    const response = await fetch(url, {
-      method: 'POST',
-      body: content,
-    });
-    if (response.ok) {
-      console.log('Success:', response);
-    } else {
-      throw new Error(`Failed to post: ${url}`);
+  // For a single session, POST all of the relevant content that needs to be
+  // updated for that session.
+  // A session will remain stale if any of the POSTs fail. A stale session will
+  // be forced to send all of the content the next time the refresh button is
+  // clicked.
+  async sendSessionContent(
+      session: MobileSession, updatedContent: EditorUpdates,
+      usdzBlob: Blob|undefined, gltfBlob: Blob|undefined,
+      envBlob: Blob|undefined) {
+    if (session.isStale) {
+      updatedContent = this.getStaleContent();
     }
-  }
+    session.isStale = true;
 
-  // Create object to tell mobile what information is being sent so it can
-  // dynamically send certain GET requests
-  getUpdatedContent() {
-    return {
-      gltfChanged: this.isNewModel(), stateChanged: this.stateHasChanged(),
-          envChanged: this.isNewSource(this.urls.env, this.lastUrlsSent.env),
-          envIsHdr: this.envIsHdr(), modelIds: this.getRandomInt(1e+20),
-          iosChanged: this.isNewSource(this.urls.usdz, this.lastUrlsSent.usdz)
+    const packet: MobilePacket = {updatedContent: updatedContent};
+    if (updatedContent.stateChanged) {
+      packet.snippet = this.snippet;
+    }
+
+    await post(JSON.stringify(packet), getSessionUrl(this.pipeId, session.id));
+
+    if (session.os === 'iOS' && updatedContent.iosChanged && usdzBlob) {
+      await post(
+          usdzBlob,
+          usdzToSession(this.pipeId, session.id, updatedContent.usdzId));
     }
+
+    if (updatedContent.gltfChanged && gltfBlob) {
+      await post(
+          gltfBlob,
+          gltfToSession(this.pipeId, session.id, updatedContent.gltfId));
+    }
+
+    if (updatedContent.envChanged && envBlob) {
+      await post(envBlob, envToSession(this.pipeId, session.id));
+    }
+
+    // The isStale flag will stay true if all of the requests are not delivered.
+    session.isStale = false;
   }
 
-  // Send any state, model, or image that has been updated since the last update
+  // Send any state, model, or environment iamge that has been updated since the
+  // last refresh.
+  // TODO: Now that the posts are asynchronous, need to find a new way to deal
+  // with knowing when isSendingData should be set to false.
```

Comment:
How will this problem manifest?

Reply:
Users would just have the ability to continuously press the refresh button instead of there being a delay in that ability as the content is being loaded.